sdk/storage/azblob/blob/client.go

```diff
@@ -387,14 +387,17 @@ func (b *Client) downloadBuffer(ctx context.Context, writer io.WriterAt, o downl
 			if err != nil {
 				return err
 			}
+			progressLock.Lock()
```

Comment:
If we have computed the length of data using getProperties then we shall not be taking cost of locking and incrementing the variables here.